diff --git a/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm b/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm
index da42873..c2d4915 100644
--- a/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm
+++ b/node_modules/react-native-date-picker/ios/RNDatePickerManager.mm
@@ -7,6 +7,10 @@
 #import "DatePicker.h"
 #import "RNDatePicker.h"
 
+#ifdef RCT_NEW_ARCH_ENABLED
+#import "RNDatePickerSpecs.h"
+#endif
+
 @implementation RCTConvert(UIDatePicker)
 
 RCT_ENUM_CONVERTER(UIDatePickerMode, (@{
@@ -22,6 +26,14 @@ @implementation RNDatePickerManager
 
 RCT_EXPORT_MODULE(RNDatePicker)
 
+#ifdef RCT_NEW_ARCH_ENABLED
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+    return std::make_shared<facebook::react::NativeRNDatePickerSpecJSI>(params);
+}
+#endif
+
 RCT_EXPORT_METHOD(addListener : (NSString *)eventName) {
   // Keep: Required for RN built in Event Emitter Calls.
 }
